examples/vision/attention_mil_classification.py

```diff
@@ -0,0 +1,575 @@
+"""
+Title: Classification using Attention-based Deep Multiple Instance Learning (MIL).
+Author: [Mohamad Jaber](https://www.linkedin.com/in/mohamadjaber1/)
+Date created: 2021/08/16
+Last modified: 2021/09/09
+Description: MIL approach to classify bags of instances and get their individual instance score.
+"""
+"""
+Suppose there is a locked door and several key chains with each chain containing few keys.
+The goal here is to be able to know which key chain (in general) and key (in specific) can
+open up the locked door.
+
+In order to solve this task, we shall find the key that can opens the door. If the key was
+correctly found, then the entire key chain will be classified as positive. If the key chain
+doesn't contain any compatible key, it will be classified as negative.
+
+In the context of MIL, each of the key chains are known as **bags**, in which each can be
+labelled as positive or negative. Each key within the key chains are known as **instances**.
+
+## Introduction
+
+Usually for image classification tasks using **supervised learning**  algorithms, the learner
+receives a single label for a single image. In case of **MIL**, the learner receives a single
+label for a set of images.
+
+The motivation behind this model is to:
```

Comment:
We need to provide stronger examples / use cases to justify the interestingness of this setup.

Reply:
Added.